src/lib/utils/__tests__/enhanceConfiguration-test.js

```diff
@@ -85,4 +90,20 @@ describe('enhanceConfiguration', () => {
       })
     );
   });
+
+  it('does not duplicate hierarchicalFacets (object in array)', () => {
```

Comment:
this test failed with the previous implementation and is added to prove #3278 is fixed